x-pack/plugins/spaces/server/config.ts

```diff
@@ -26,11 +26,9 @@ export const ConfigSchema = schema.object({
     })
   ),
   maxSpaces: schema.number({ defaultValue: 1000 }),
-  allowFeatureVisibility: schema.conditional(
-    schema.contextRef('serverless'),
-    true,
-    schema.literal(false),
-    schema.boolean({
+  allowFeatureVisibility: offeringBasedSchema({
```

Comment:
__note:__ I believe we added this flag before `schema.contextRef('serverless')` was implemented, which is why it's a bit verbose. So if I'm not mistaken, we can simplify it down to the snippet below. But, we'll need to backport this change to 8.10 to not release previous version, so feel free to leave it as it is, we can make adjustments in a subsequent PR. cc @elastic/kibana-security 
```ts
  allowFeatureVisibility: offeringBasedSchema({
    serverless: schema.literal(false),
  }),
```